### ISSUE TITLE
2015-09-04-iterating-through-lists-of-lists-of-lists

### ISSUE THREAD

**Issue by mkoohafkan** [mkoohafkan/mkoohafkan.github.io#20]
comments on http://hydroecology.net/iterating-through-lists-of-lists-of-lists/

**Comment by Nafis** [comment]
Hi, I am writting my own programme in R about the segmentation tree, I have lots of challenge in making a list from my recursive function (this function should calculate my tree node's information), I have found your page and I guess that maybe you have experiences in writting recursive function in R and maybe you can help me. Thanks en advance

**Comment by mkoohafkan** [comment]
Hmm, I've never worked with segmentation trees before. @Nafis, have you searched or posted a question on [stack overflow](https://stackoverflow.com)?

**Comment by Nafis** [comment]
No, not yet!  But it's not really about the segmentation trees;

for example

I wrote a function  takes a vector (x=1:100) and in each step it divides the vector into two vector(x<50 and x>50)

but I cant allocate a list to the obtained informations in each step!




 test<-function(x,p,l=list()){

  print('____________________________')

  if(length(x)>2 & p<4){



  xg<-x[x<(min(x)+max(x))/2]
  xd<-x[x>(min(x)+max(x))/2]
  p=p+1
  d<-list(xg,xd)

  for(j in 1:length(d)){
    print(paste(p,"------>",p-1))
    print(paste(min(x),max(x)))
    l<-test(d[[j]],p,l)
    }


  }
   return(l)
}


result<-test(1:50,0,l=list())



As we see, the result won't give me the entire calculation! You may have a solution!!??




________________________________

**Comment by mkoohafkan** [comment]
Definitely post this sort of thing on Stack Overflow, you'll get more responses that way :)
